File: ClinicSystem/src/com/example/clinicsystem/models/classes/M_Person.java
package com.example.clinicsystem.models.classes;

import com.example.clinicsystem.helpers.DatabaseConnection;
import com.example.clinicsystem.models.enums.Gender;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class M_Person {
    private String firstName, middleName, lastName, phoneNumber, birthDate;
    private M_Address address;
    private int nationalID;
    private Gender gender;

    public M_Person(String firstName, String middleName, String lastName, String phoneNumber, String birthDate, M_Address address, int nationalID, Gender gender) {
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
        this.phoneNumber = phoneNumber;
        this.birthDate = birthDate;
        this.address = address;
        this.nationalID = nationalID;
        this.gender = gender;
    }

    public M_Person() {
        this.address = new M_Address();
    }

    public int addAddress() {
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();

        return databaseConnection.insertInto("address","(Address_City, Address_Street," +
                "Address_Apartment, Address_Governorate)" ,"('" + address.getCity() + "', '"
                + address.getStreetName() + "', '" + address.getApartmentNumber() + "', '" + address.getGovernorate()
                + "')");
    }

    public List<M_Address> getAllAddresses(String condition) {
        List<M_Address> addresses = new ArrayList<>();
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();
        ResultSet resultSet = databaseConnection.select("address", condition);

        if(resultSet != null) {
            try {
                while (resultSet.next()) {
                    int id = resultSet.getInt("Address_ID");
                    String apt = resultSet.getString("Address_Apartment");
                    String city = resultSet.getString("Address_City");
                    String gover = resultSet.getString("Address_Governorate");
                    String st = resultSet.getString("Address_Street");

                    M_Address address = new M_Address(id, city, st, gover, apt);
                    addresses.add(address);
                }
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

        return addresses;
    }

    public int addPerson() {
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();
        int genderID;

        if (this.gender == Gender.MALE) {
            genderID = 1;
        } else {
            genderID = 2;
        }

        return databaseConnection.insertInto("person","(Address_ID, Gender_ID, Person_Birth" +
                ", Person_FN, Person_LN, Person_MN, Person_NationalID, Person_Phone)"
                ,"('" + this.address.getId() + "', '" + genderID + "', '" + this.birthDate + "', '" + this.firstName + "', '"
                + this.lastName + "', '" + this.middleName + "', '" + this.nationalID + "', '" + this.phoneNumber + "')");
    }

    public List<M_Person> getAllPersons(String condition) {
        List<M_Person> people = new ArrayList<>();
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();
        ResultSet resultSet = databaseConnection.select("person", condition);

        if(resultSet != null) {
            try {
                while (resultSet.next()) {
                    String fn = resultSet.getString("Person_FN");
                    String mn = resultSet.getString("Person_MN");
                    String ln = resultSet.getString("Person_LN");
                    String dob = resultSet.getString("Person_Birth");
                    int id = resultSet.getInt("Person_NationalID");
                    String phone = resultSet.getString("Person_Phone");
                    Gender gender;

                    if(resultSet.getInt("Gender_ID") == 1) {
                        gender = Gender.MALE;
                    } else {
                        gender = Gender.FEMALE;
                    }

                    List<M_Address> addresses = getAllAddresses(" WHERE Address_ID = " + resultSet.getInt("Address_ID"));

                    M_Person person = new M_Person(fn, mn, ln, phone, dob, addresses.get(0), id, gender);
                    people.add(person);
                }
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

        return people;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public void setMiddleName(String middleName) {
        this.middleName = middleName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(String birthDate) {
        this.birthDate = birthDate;
    }

    public M_Address getAddress() {
        return address;
    }

    public void setAddress(M_Address address) {
        this.address = address;
    }

    public int getNationalID() {
        return nationalID;
    }

    public void setNationalID(int nationalID) {
        this.nationalID = nationalID;
    }

    public Gender getGender() {
        return gender;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }
}

File: clinic/M_Person.php
<?php
include_once "M_Address.php";
include_once "DatabaseConnection.php";

class M_Person {
	private  $firstName, $middleName, $lastName, $phoneNumber, $birthDate;
    private  $address; //Object from M_Address.
    private  $nationalID;
    private  $gender; //Enum Gender.
	
	public function __construct($firstName, $middleName, $lastName, $phoneNumber, $birthDate
	, $address, $nationalID, $gender) {
		$this->firstName = $firstName;
		$this->middleName =$middleName;
		$this->lastName =  $lastName;
		$this->phoneNumber =$phoneNumber;
		$this->birthDate =  $birthDate;
		$this->address =    $address;
		$this->nationalID = $nationalID;
		$this->gender = 	$gender;
	}
	
	public function addAddress() {
		$conn = DatabaseConnection::getINSTANCE();
		
		return $conn->insertInto("address", "(Address_City, Address_Street, Address_Apartment, Address_Governorate)", "('". 
		$this->address->getCity() . "', '" . $this->address->getStreetName() . "', '" . $this->address->getApartmentNumber() .
		"', '" . $this->address->getGovernorate() . "')");
	}
	
	public function getAllAddresses($condition) {
        $conn = DatabaseConnection::getINSTANCE();
		
		$result = $conn->select("address", $condition);
		$addresses = array();
		
		while ($row = $result->fetch_assoc()) {
			$address = new M_Address($row["Address_ID"], $row["Address_City"]
				,$row["Address_Street"],$row["Address_Governorate"],$row["Address_Apartment"]);
			array_push($addresses, $address);
		}
		
		return $addresses;
    }

	public function getAllPersons($condition) {
		$conn = DatabaseConnection::getINSTANCE();
		
		$result = $conn->select("person", $condition);
		$people = array();
		
		if($result != null) {
			while ($row = $result->fetch_assoc()) {
				$addresses = $this->getAllAddresses(" WHERE Address_ID = " . 
					$row["Address_ID"]);
					
					$gender = "Female";
					
					if($row["Gender_ID"] == 1) {
						$gender = "Male";
					}
				
				$person = new M_Person($row["Person_FN"], $row["Person_MN"]
					, $row["Person_LN"],$row["Person_Phone"], $row["Person_Birth"]
					, $addresses[0], $row["Person_NationalID"], $gender);
				array_push($people, $person);
			}
		}
		
		return $people;
	}
	
	public function addPerson() {
		$conn = DatabaseConnection::getINSTANCE();
		
		$genderID = 1;
		
		if($this->gender == "female") {
			$genderID = 2;
		}
		
		return $conn->insertInto("person", "(Person_FN, Person_MN, Person_LN, Person_Phone,
			Person_Birth, Person_NationalID, Address_ID, Gender_ID)", "('" . $this->firstName . "', '" .
			$this->middleName . "', '" . $this->lastName . "', '" . $this->phoneNumber . "', '" .
			$this->birthDate . "', '" . $this->address->getID() . "', '" . $this->nationalID . "', '" . 
			$genderID . "')");
	}
	
	public function getFirstName() {
        return $this->firstName;
    }

    public function setFirstName($firstName) {
        $this->firstName = $firstName;
    }

    public function getMiddleName() {
        return $this->middleName;
    }

    public function setMiddleName($middleName) {
        $this->middleName = $middleName;
    }

    public function getLastName() {
        return $this->lastName;
    }

    public function setLastName($lastName) {
        $this->lastName = $lastName;
    }

    public function getPhoneNumber() {
        return $this->phoneNumber;
    }

    public function setPhoneNumber($phoneNumber) {
        $this->phoneNumber = $phoneNumber;
    }

    public function getBirthDate() {
        return $this->birthDate;
    }

    public function setBirthDate($birthDate) {
        $this->birthDate = $birthDate;
    }

    public function getAddress() {
        return $this->address;
    }

    public function setAddress($address) {
        $this->address = $address;
    }

    public function getNationalID() {
        return $this->nationalID;
    }

    public function setNationalID($nationalID) {
        $this->nationalID = $nationalID;
    }

    public function getGender() {
        return $this->gender;
    }

    public function setGender($gender) {
        $this->gender = $gender;
    }



}



?>